### PR TITLE
Moving Release Wiki to react-native-website

### PR DESCRIPTION
**Pull request by Megatron4537**
The release wiki can now be found under the community tab

A few things to note:

- The page url changed from  **/help** to **/docs/community**. We can change the suffix but for the page to have a left nav bar it needs the prefix **docs**
- The page release.md becomes redundant we could consider removing it.

[#33014](https://github.com/facebook/react-native/issues/33014)

### CODE REVIEW

File: website/versioned_sidebars/version-0.67-sidebars.json
Comment (Simek) [inline website/versioned_sidebars/version-0.67-sidebars.json:230]:
```suggestion
    "Release Processes/Guides": [
```

Not sure which fit better here, but let's do not use "wiki".

File: website/versioned_docs/version-0.67/updating-upgrade-helper.md
Comment (Simek) [inline website/versioned_docs/version-0.67/updating-upgrade-helper.md:20]:
```suggestion
- The GitHub action will be triggered by the [CircleCI job that pushes the release link](https://github.com/facebook/react-native/blob/main/.circleci/config.yml#L822).
```

There are more typos like those, it looks like the content need to be proof-read before shipping.

**Review by Simek — CHANGES_REQUESTED** [review]
Hello @Megatron4537, thank you for picking this task up! 👍 

I have take a quick glance on the additions and I have left several question and suggestions. Besides those, we definitely will need to have a formatting and proof-reading pass on the new content, since generally those wiki pages were mostly used internally (and people without the permission in main repo were not able to edit or correct those pages) and now they will be exposed to the larger community.

If you have any questions, feel free to ask them here.

File: website/versioned_docs/version-0.67/release-testing.md
Comment (Simek) [inline website/versioned_docs/version-0.67/release-testing.md:19]:
```suggestion
- Have Android and iOS development environment set-up. Follow instructions for macOS/iOS and macOS/Android from the [Environment Setup](/docs/environment-setup) guide.
```

We should use the internal links to avoid re-render on navigation.

**Comment by lunaleaps** [comment]
> I'm not sure even if we want to version those pages anyway (CC @lunaleaps)

Hmm, my instinct is to not have these versioned but I do think there are instruction changes in different versions. Like adding new automation etc. for a new release candidate but I do doubt if it's intuitive to see release instructions for different versions? Maybe? Would love to hear others' thoughts

**Comment by cortinico** [comment]
> Hmm, my instinct is to not have these versioned but I do think there are instruction changes in different versions

+1 on this. I'm not sure how we can separate some docs from being not versioned, but it will definitely reduce the maintenance costs.

**Comment by lunaleaps** [comment]
@Megatron4537 Is this something you still want to tackle? Let me know if you need help or need to step away from it

**Comment by Megatron4537** [comment]
> @Megatron4537 Is this something you still want to tackle? Let me know if you need help or need to step away from it

@lunaleaps Yes, I am just unsure of how to approach it. Which pages should be versioned and which shouldn't be? The implementation is different for versioned and not versioned

**Comment by lunaleaps** [comment]
@Megatron4537 I think your changes are un-versioned right now which is how we want it.

File: website/src/css/customTheme.scss
Comment (lunaleaps) [reply website/src/css/customTheme.scss:1433]: Agreed, would be great to have the "new" label on the group

**Comment by Megatron4537** [comment]
@lunaleaps I made the changes with all the pages but a new sidebar if you want to have a look and/or proof-read. We can also opt to have only a few pages to begin with if you think it's more appropriate.

@Simek I made all requested changes except the "New" label for the release group which I don't know how to do with Docusaurus' template.

Let me know if this is what you two had in mind for the sidebar and release process page breakdown.

The only thing untouched is I couldn't add the "New" label to the group with the Docusaurus template.

**Comment by Simek** [comment]
@Megatron4537 Thank you for the updates, but you are still working on the `0.67` versioned docs files. To make the pages "unversioned", and detached form the versioning mechanism, you need to place files in `website/src/pages` directory, you can check out the details here:
* https://docusaurus.io/docs/next/creating-pages#add-a-markdown-page

> Let me know if this is what you two had in mind for the sidebar and release process page breakdown.

The sidebar organization definitely looks better now, it is also something that we can iterate and improve. We need a bit more consistency in the sidebar names (ex. "release/releasing", unified capitalization), also the header to the releases guides can be improved.

> The only thing untouched is I couldn't add the "New" label to the group with the Docusaurus template.

"New" label is not that important right now, we can add it later, in the separate PR. It's a custom CSS-based solution, which is not a part of Docusaurus by default so don't worry about that.

**Comment by Megatron4537** [comment]
@Simek Allright, thanks for the quick review!

I understand for the unversioned docs but I can't use the link you referred for this since Pages don't allow use sidebars. Only docs can use sidebars

**Comment by Simek** [comment]
> I understand for the unversioned docs but I can't use the link you referred for this since Pages don't allow use sidebars. Only docs can use sidebars

@Megatron4537 Docusaurus offers an ability to group the custom pages in the custom sidebars, you can check more details in here:
* https://docusaurus.io/docs/next/sidebar/multiple-sidebars#sidebar-association

**Comment by Megatron4537** [comment]
@Simek I moved all community pages to their own subfolder. Let me know if there's anything else to do :)

**Comment by Simek** [comment]
@Megatron4537 Thank you for all the updates! 👍 

Docs structure looks fine now, we can improve it later, it's time to focus on the page naming, page content and formatting.

If you visit the deploy preview, you can see that there are several minor rendering and formatting issues like broken tables, code block with additional whitespace, duplicated headers or missing punctuation marks.

I will try to leave few suggestions later today, but it would be nice to have the native speaker to proof-read those docs.

**Comment by kelset** [comment]
thanks for doing the work @Megatron4537! I think it's a great starting point to have a bit of a broader conversation - I feel that having the release wiki stuff under the "community" name is super confusing (but it's not your fault!).

So, right now we have a situation with, in the sidebar:
* Guides
* Components
* API
* Architecture
* Community
* Blog

And the community page "Support" is very, very misleading as a name.

So here are a couple of small tweaks that I'd like to propose.

1. rename the top section "Community" to "Contributing".
    1. "Community Overview" -> "Contributing to React Native"
    1. "Support" page becomes like "How to contribute" landing page saying something like thanks that you want to help, please head over to CONTRIBUTING.md (or migrate the contributing.md here, and have the markdown be the "thanks for contributing, head here for info")
1. the "community page" from being `/help` becomes `/community`, stays as-is and it's only linked via the footer, just like for the showcase

I think that would set everything in a better state. I'm not saying all of it needs to be done in this PR (actually, it'd be probably better to do the above as a separate PR, then rebase this one on top of it)

**Comment by Megatron4537** [comment]
I made this new alternative PR https://github.com/facebook/react-native-website/pull/2989 with your requested changes @kelset 

Let me know if there's other changes you'd like

**Comment by kelset** [comment]
> What's the reason for an additional and separate PR? It is a bit confusing, and makes it harder to suggest further changes.

@Simek it's on me - I proposed to do as a separate PR to make it easier to review in isolation :) Sorry, I'm a person that prefers that kind of "dividi et impera" approach over having 1 massive PR 😅

But yeah it looks like @Megatron4537 still kept in that other PR all the release wiki changes too which kinds of defeats the purpose of having as two separate PRs 😅😅😅 @Megatron4537 can you remove the "release wiki" stuff from the other PR?

**Comment by Megatron4537** [comment]
> > I made this new alternative PR #2989 with your requested changes @kelset
> 
> What's the reason for an additional and separate PR? It is a bit confusing, and makes it harder to suggest further changes.
> 
> Also I would like to contribute to your PR during weekend and focus on the formatting and wording, is it fine for you?

Yes it's totally fine!


> > What's the reason for an additional and separate PR? It is a bit confusing, and makes it harder to suggest further changes.
> 
> @Simek it's on me - I proposed to do as a separate PR to make it easier to review in isolation :) Sorry, I'm a person that prefers that kind of "dividi et impera" approach over having 1 massive PR sweat_smile
> 
> But yeah it looks like @Megatron4537 still kept in that other PR all the release wiki changes too which kinds of defeats the purpose of having two separate PRs sweat_smilesweat_smilesweat_smile @Megatron4537 can you remove the "release wiki" stuff from the other PR? Or just have all the changes in this PR (and close the other) - either one is fine :)

:joy: Ok it was easier for me to merge everything in this PR. I closed the other PR. Sorry if it makes reviewing a bit harder, I will make a note to have smaller incremental PRs in the future.

**Comment by kelset** [comment]
Just a quick check in - @Simek & @lunaleaps from your perspective, aside the feedbacks Simek has already provided, is this in good shape? Personally I think it's 👍

**Comment by lunaleaps** [comment]
Yea this is looking good to me!

**Comment by lunaleaps** [comment]
I'll let @Simek reply and merge if it's good, thanks @Megatron4537 for pushing forward on this! It'll be great to have all this information more visible!

**Comment by Simek** [comment]
Since I have updated the page names and structure I will re-request the review.

Also most of the formatting changes and typos were fixed, but I still think that it would be nice to have a pass over (and potentially rephrase) the content of some pages by native speaker.

There are still sections or places which seems to be WIP or not finished, for example the "Dependencies" table comments:
> Needs to be clarified (kept up to date by internal team?).

It would be nice to update this, but I think it's not a shipping blocker.

File: website/contributing/release-candidate-minor.md
Comment (Simek) [inline website/contributing/release-candidate-minor.md:153]:
```suggestion
[changelog-wiki]: https://github.com/facebook/react-native/wiki/Release-Changelog
```

It looks like this wiki page do not exist anymore, and link leads to wiki home. Any idea if this has this been moved somewhere else @lunaleaps? Maybe releases WG repo?

Comment (Simek) [reply website/contributing/release-candidate-minor.md:153]: This what is the message I see when I try to access the link, which is a bit confusing:
Screenshot 2022-03-08 at 13 16 31

File: website/docusaurus.config.js
Comment (slorber) [inline website/docusaurus.config.js:158]: useless if the plugin is not versioned

File: website/docusaurus.config.js
Comment (slorber) [inline website/docusaurus.config.js:159]:
```suggestion
        sidebarPath: require.resolve('./sidebarsContributing.json'),
```

For consistency?

File: website/docusaurus.config.js
Comment (slorber) [inline website/docusaurus.config.js:155]: Some options of the main versioned docs instance are missing, is this on purpose?

```js
        showLastUpdateAuthor: false,
        showLastUpdateTime: true,
        editUrl:
          'https://github.com/facebook/react-native-website/blob/master/website/',
        remarkPlugins: [require('@react-native-website/remark-snackplayer')],
```

As I'm also adding another docs instance in https://github.com/facebook/react-native-website/pull/3000, maybe it should later be refactored so that those docs options are shared over instances?

File: website/docusaurus.config.js
Comment (Simek) [reply website/docusaurus.config.js]: Fixed for now in latest commit, but we can adjust this once more later, to merge this with approach in the architecture PR.

File: website/docusaurus.config.js
Comment (Simek) [reply website/docusaurus.config.js]: Done! 👍

**Review by lunaleaps — APPROVED** [review]
lgtm! I think it's fine if we have some things in typo/unfilled state. We can continue to iterate on smaller PRs to fix.

**Review by kelset — COMMENTED** [review]
🚢

(after merging this I guess we could rebase https://github.com/facebook/react-native-website/pull/3000 to also unversion this section, right?)

**Comment by Simek** [comment]
> 🚢
> 
> (after merging this I guess we could rebase #3000 to also unversion this section, right?)

Heh, I have planned the reversed merge order for this, we are currently finishing the other PR with Sebastien, and we would stay with two additional docs plugins to simplify the paths. 🙂

**Comment by kelset** [comment]
> Heh, I have planned the reversed merge order for this, we are currently finishing the other PR with Sebastien, and we would stay with two additional docs plugins to simplify the paths. 🙂

Sounds good 👍

**Comment by Simek** [comment]
Let's change the plans and ship his one first, since we have few small thing to adjust in the New Arch PR, and this one looks ready to merge as-is right now (and there should not be that many conflicts, if any).

I would like to thank @Megatron4537 for the all the work they done in there, that's awesome to have such involved community members like you! ❤️

**Review by Simek — APPROVED** [review]

**Pull request merged**
